warehouse/query-core/src/main/java/datawave/next/DocIdIteratorVisitor.java

```diff
@@ -0,0 +1,521 @@
+
+        // now process negations
+        for (JexlNode child : negative) {
+            // intersections drive their own context
+            Object o = child.jjtAccept(this, result);
+            if (!(o instanceof ScanResult)) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Node did not return a set: {}", JexlStringBuildingVisitor.buildQuery(child));
+                }
+                continue;
+            }
+
+            ScanResult scanResult = (ScanResult) o;
+            if (scanResult.getResults().isEmpty()) {
+                if (log.isDebugEnabled()) {
+                    log.debug("negated term in intersection, child returned zero hits");
+                }
+                continue;
+            }
+
+            // uncomment for exceptions
+            // Preconditions.checkNotNull(ids);
+            if (result != null && !result.getResults().isEmpty()) {
+                result.getResults().removeAll(scanResult.getResults());
+            }
+
+            if (result != null && result.getResults().isEmpty()) {
+                if (log.isDebugEnabled()) {
+                    log.debug("no ids exist for intersection after processing merge, short circuit return");
+                }
+                return result;
+            }
+        }
+
+        if (result == null) {
+            // no terms were executable
+            if (log.isDebugEnabled()) {
+                log.debug("intersection: [{}] found 0 hits", JexlStringBuildingVisitor.buildQuery(node));
+            }
+            return data;
+        }
+        if (log.isDebugEnabled()) {
+            log.debug("intersection: [{}] found {} hits", JexlStringBuildingVisitor.buildQuery(node), result.getResults().size());
+        }
+        return result;
+    }
+
+    /**
+     * This method exists because we may have a bounded range that is also marked as value exceeded
+     *
+     * @param node
+     *            the original ASTAndNode
+     * @param data
+     *            the data
+     * @param instance
+     *            the QueryPropertyMarker Instance
+     * @return an object
+     */
+    private Object handleMarker(ASTAndNode node, Object data, QueryPropertyMarker.Instance instance) {
+        switch (instance.getType()) {
+            case BOUNDED_RANGE:
+                return handledBoundedRange(node, data, instance);
+            case EXCEEDED_OR:
+                return handleListMarker(node, data, instance);
+            case EXCEEDED_VALUE:
+                return handleExceededValue(node, data, instance);
+            case INDEX_HOLE:
+                log.info("found an index hole");
+                return data;
+            case EVALUATION_ONLY:
+            case DELAYED:
+            case EXCEEDED_TERM:
+            case DROPPED:
+            case STRICT:
+            case LENIENT:
+                log.debug("not handling marker of type: {}", instance.getType().getLabel());
```

Comment:
should these throw an error since they won't be handled? Or is it expected they will be handled later in the evaluation just not yet?

Reply:
These are markers that are not handled today, and in most cases they should be removed from the query during planning. I wanted to record the existence of these markers in case we ever miss something. 

I don't think it's worth outright failing the query.